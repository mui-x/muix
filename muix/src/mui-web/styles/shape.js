//----------------------------------------------------------------------------------
//
// This code was generated from material-ui v1.4.2 by reactxx-codemod tool
// (https://github.com/reactxx/reactxx/tree/master/codemod)
//
//----------------------------------------------------------------------------------

const shape = {
  borderRadius: 4
};
export default shape;
