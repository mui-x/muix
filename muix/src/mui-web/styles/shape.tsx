//autogenerated--------------------------------------------------------------------
// 
// This code was generated from material-ui v3.0.0 by reactxx-codemod tool
// (https://github.com/reactxx/reactxx/tree/master/codemod)
// 
//----------------------------------------------------------------------------------

export interface Shape {
  borderRadius: number;
}
export type ShapeOptions = Partial<Shape>;
const shape = {
  borderRadius: 4
};
export default shape;
