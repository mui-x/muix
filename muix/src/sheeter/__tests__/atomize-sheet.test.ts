import { ts, Shape } from "./shape";
import { atomizeSheet } from "reactxx-sheeter";
import { initPlatform } from "reactxx-tests";

describe("core sheet simple", () => {
  const createSheet = () =>
    (ts.sheet = {
      root: (ts.view = {
        margin: 4,
        padding: 4,
        $web: (ts.web = {
          margin: 10
        }),
        $native: (ts.nativeView = {
          padding: 10
        })
      }),
      label: (ts.text = [
        {
          color: "blue"
        }
      ]),
      webOnly: {
        $web: (ts.web = {
          color: "red"
        })
      },
      nativeOnly: {
        $native: (ts.nativeView = {
          margin: 20
        })
      }
    });
  it("NATIVE, DEVELOPMENt", () => {
    initPlatform(false);
    const atomizedSheet = atomizeSheet<Shape>(createSheet(), null);
    expect(atomizedSheet).toMatchInlineSnapshot(`
Object {
  "label": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "color: blue /*label[0]*/",
        ],
      },
    ],
    "name": "label",
    "~": "c",
  },
  "nativeOnly": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "margin: 20 /*nativeOnly/$native*/",
        ],
      },
    ],
    "name": "nativeOnly",
    "~": "c",
  },
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "margin: 4 /*root*/",
          "padding: 4 /*root*/",
        ],
      },
      Object {
        "atomicArray": Array [
          "padding: 10 /*root/$native*/",
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
}
`);
  });
  it("NATIVE, PRODUCTION", () => {
    initPlatform(false, false);
    const atomizedSheet = atomizeSheet<Shape>(createSheet(), null);
    expect(atomizedSheet).toMatchInlineSnapshot(`
Object {
  "label": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          Object {
            "propId": "color",
            "value": "blue",
          },
        ],
      },
    ],
    "name": "label",
    "~": "c",
  },
  "nativeOnly": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          Object {
            "propId": "margin",
            "value": 20,
          },
        ],
      },
    ],
    "name": "nativeOnly",
    "~": "c",
  },
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          Object {
            "propId": "margin",
            "value": 4,
          },
          Object {
            "propId": "padding",
            "value": 4,
          },
        ],
      },
      Object {
        "atomicArray": Array [
          Object {
            "propId": "padding",
            "value": 10,
          },
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
}
`);
  });
  it("WEB DEVELOPMENT", () => {
    initPlatform(true);
    const atomizedSheet = atomizeSheet<Shape>(createSheet(), null);
    expect(atomizedSheet).toMatchInlineSnapshot(`
Object {
  "label": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".d { color:blue /*label[0]*/ }",
        ],
      },
    ],
    "name": "label",
    "~": "c",
  },
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".a { margin:4px /*root*/ }",
          ".b { padding:4px /*root*/ }",
        ],
      },
      Object {
        "atomicArray": Array [
          ".c { margin:10px /*root/$web*/ }",
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
  "webOnly": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          ".e { color:red /*webOnly/$web*/ }",
        ],
      },
    ],
    "name": "webOnly",
    "~": "c",
  },
}
`);
  });
  it("WEB PRODUCTION", () => {
    initPlatform(true, false);
    const atomizedSheet = atomizeSheet<Shape>(createSheet(), null);
    expect(atomizedSheet).toMatchInlineSnapshot(`
Object {
  "label": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "d",
        ],
      },
    ],
    "name": "label",
    "~": "c",
  },
  "root": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "a",
          "b",
        ],
      },
      Object {
        "atomicArray": Array [
          "c",
        ],
      },
    ],
    "name": "root",
    "~": "c",
  },
  "webOnly": Object {
    "list": Array [
      Object {
        "atomicArray": Array [
          "e",
        ],
      },
    ],
    "name": "webOnly",
    "~": "c",
  },
}
`);
  });
});
