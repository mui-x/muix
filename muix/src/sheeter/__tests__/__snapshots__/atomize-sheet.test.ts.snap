// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ATOMIZE SHEET NATIVE, DEVELOPMENt 1`] = `
Object {
  "label": Array [
    Array [
      "color: blue /*sheet.label[0]*/",
    ],
  ],
  "nativeOnly": Array [
    Array [
      "margin: 20 /*sheet.nativeOnly/$native*/",
    ],
  ],
  "root": Array [
    Array [
      "margin: 4 /*sheet.root*/",
      "padding: 4 /*sheet.root*/",
    ],
    Array [
      "padding: 10 /*sheet.root/$native[0]*/",
    ],
  ],
}
`;

exports[`ATOMIZE SHEET NATIVE, PRODUCTION 1`] = `
Object {
  "label": Array [
    Array [
      Object {
        "propId": "color",
        "value": "blue",
      },
    ],
  ],
  "nativeOnly": Array [
    Array [
      Object {
        "propId": "margin",
        "value": 20,
      },
    ],
  ],
  "root": Array [
    Array [
      Object {
        "propId": "margin",
        "value": 4,
      },
      Object {
        "propId": "padding",
        "value": 4,
      },
    ],
    Array [
      Object {
        "propId": "padding",
        "value": 10,
      },
    ],
  ],
}
`;

exports[`ATOMIZE SHEET WEB DEVELOPMENT 1`] = `
Object {
  "label": Array [
    Array [
      ".d { color:blue /*sheet.label[0]*/ }",
    ],
  ],
  "root": Array [
    Array [
      ".a { margin:4px /*sheet.root*/ }",
      ".b { padding:4px /*sheet.root*/ }",
    ],
    Array [
      ".c { margin:10px /*sheet.root/$web*/ }",
    ],
  ],
  "webOnly": Array [
    Array [
      ".e { color:red /*sheet.webOnly/$web[0]*/ }",
    ],
  ],
}
`;

exports[`ATOMIZE SHEET WEB PRODUCTION 1`] = `
Object {
  "label": Array [
    Array [
      "d",
    ],
  ],
  "root": Array [
    Array [
      "a",
      "b",
    ],
    Array [
      "c",
    ],
  ],
  "webOnly": Array [
    Array [
      "e",
    ],
  ],
}
`;
